x/staking/migrations/v5/keys.go

```diff
@@ -60,16 +60,27 @@ func ParseDelegationKey(bz []byte) (sdk.AccAddress, sdk.ValAddress, error) {
 func GetHistoricalInfoKey(height int64) []byte {
 	heightBytes := make([]byte, 8)
 	binary.BigEndian.PutUint64(heightBytes, uint64(height))
-	return append(HistoricalInfoKey, heightBytes...)
+	key := make([]byte, len(HistoricalInfoKey)+len(heightBytes))
+	copy(key, HistoricalInfoKey)
+	copy(key[len(HistoricalInfoKey):], heightBytes)
+	return key
```

Comment:
[no change needed] I think this could be refactored

```suggestion
	return slices.Concat(HistoricalInfoKey, heightBytes)
```

- it creates a new slice that is allocated to the length of len(HistoricalInfoKey)+len(heightBytes) (actually slightly b/c `Grow`)
- it isn't as verbose
- it's less likely to result in an indexing error. To me this looks sketchy even though it is correct

```
	copy(key[len(HistoricalInfoKey):], heightBytes)
```

Ref: https://cs.opensource.google/go/go/+/refs/tags/go1.24.4:src/slices/slices.go;l=478-495